core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -146,6 +151,7 @@ public Socket connect(ConnectionConfig config) throws IOException, InterruptedEx
   }
 
   public ConnectionMetadata getConnectionMetadata(ConnectionConfig config) throws IOException {
+    config = resolveConfig(config);
```

Comment:
Is it required here?

`resolveConfig()` is called inside `getConnector()` as well.

Reply:
I've updated this and also `connect()` to hopefully be more straightforward.